frontend/src/modules/scaffold/views/scaffoldAssetItemView.js

```diff
@@ -0,0 +1,221 @@
+define([
+  'core/origin',
+  'backbone-forms',
+  'core/helpers',
+  './scaffoldAssetView',
+  'core/models/courseAssetModel',
+  'modules/assetManagement/views/assetManagementModalView',
+  'modules/assetManagement/collections/assetCollection'
+], function(Origin, BackboneForms, Helpers, ScaffoldAssetView, CourseAssetModel, AssetManagementModalView, AssetCollection) {
+
+  var ScaffoldAssetItemView = ScaffoldAssetView.extend({
+
+    events: function() {
+      return _.extend({}, ScaffoldAssetView.prototype.events, {
+        'click .add-item': 'onAddItemClick',
+        'click .ui-draggable': 'onDragItemClick'
+      });
+    },
+
+  	onAddItemClick: function(event) {
+      event.preventDefault();
+      this.itemListView.addItem();
+    },
+
+    onDragItemClick: function(event) {
+      var item = _.find(this.dragItems, function(i) {
+        return i.$el[0] === event.currentTarget;
+      });
+      item.view.openEditor();
+    },
+
+    initialize: function(options) {
+      ScaffoldAssetView.prototype.initialize.apply(this, arguments);
+      this.init(options);
+    },
+
+    init: function(options) {
+      var attrs = options.schema.inputType;
+      this.dragItems = [];
+      this.precision = attrs.precision || 3;
+      this.topAttr = attrs.topAttribute || '_top';
+      this.leftAttr = attrs.leftAttribute || '_left';
+      this.itemsPath = attrs.items || 'properties._items';
```

Comment:
Apologies if I'm missing something obvious but what is `itemsPath` used for?

Reply:
Thanks, this must be a left over. Initially I thought it could be used to specify the path to the items attribute that is used for the pins.